presto-orc/src/test/java/com/facebook/presto/orc/reader/vector/vectorReadAndFilter/TestDoubleVectorReadAndFilter.java

```diff
@@ -0,0 +1,438 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.reader.vector.vectorReadAndFilter;
+
+import com.facebook.presto.common.predicate.vector.DoubleRangeVector;
+import com.facebook.presto.common.predicate.vector.TupleDomainFilterVector;
+import com.facebook.presto.orc.reader.vector.DoubleVectorSelectiveReadAndFilterSpecies512;
+import com.facebook.presto.orc.stream.DoubleInputStream;
+import org.mockito.Mockito;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import static org.mockito.Matchers.anyInt;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestDoubleVectorReadAndFilter
+{
+    TupleDomainFilterVector filter;
+
+    DoubleInputStream inputStream;
+
+    int[] positions;
+
+    int positionCount;
+
+    int[] outputPositions;
+
+    @BeforeMethod
+    public void setup()
+    {
+        filter = (TupleDomainFilterVector) DoubleRangeVector.of(10.0, false, false, 40.0, false, false, false);
+        inputStream = Mockito.mock(DoubleInputStream.class);
+    }
+
+    @Test
+    public void testBothVectorAndTailPart()
+    {
+        try {
+            // Both vector part and tail part of logic is executed
+            double[] doubleArr = new double[20];
+            for (int i = 1; i < 21; i++) {
+                doubleArr[i - 1] = (double) ((i * 10) % 80);
+            }
+            ByteBuffer bb = ByteBuffer.allocate(doubleArr.length * 8);
+            bb.order(ByteOrder.LITTLE_ENDIAN);
+            for (double d : doubleArr) {
+                bb.putDouble(d);
+            }
+            Mockito.when(inputStream.readDoubleDataInBytes(anyInt())).thenReturn(bb.array());
```

Comment:
we don't use mocks in presto. Please stub out any classes that you don't want to create fully.